From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 20 Aug 2026 00:14:43 +0200
Subject: popupMenu: Do not emit any active-changed signal if no submenu item
 has changed

We might have emitted the signal even for the current menu and it was
quite pointless
---
 js/ui/popupMenu.js | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/js/ui/popupMenu.js b/js/ui/popupMenu.js
index 93191a2..1d85891 100644
--- a/js/ui/popupMenu.js
+++ b/js/ui/popupMenu.js
@@ -748,7 +748,10 @@ export class PopupMenuBase extends Signals.EventEmitter {
     }
 
     _subMenuActiveChanged(submenu, submenuItem) {
-        if (this._activeMenuItem && this._activeMenuItem !== submenuItem)
+        if (this._activeMenuItem === submenuItem)
+            return;
+
+        if (this._activeMenuItem)
             this._activeMenuItem.active = false;
         this._activeMenuItem = submenuItem;
         this.emit('active-changed', submenuItem);
